clients/drcachesim/common/trace_entry.h

```diff
@@ -663,8 +699,22 @@ typedef enum {
     OFFLINE_FILE_TYPE_IFILTERED = 0x80,  /**< Instruction addresses filtered online. */
     OFFLINE_FILE_TYPE_DFILTERED = 0x100, /**< Data addresses filtered online. */
     OFFLINE_FILE_TYPE_ENCODINGS = 0x200, /**< Instruction encodings are included. */
-    /** System call number markers are included. */
+    /** System call number markers (#TRACE_MARKER_TYPE_SYSCALL) are included. */
     OFFLINE_FILE_TYPE_SYSCALL_NUMBERS = 0x400,
+    /**
+     * Kernel scheduling information is included:
+     * #TRACE_MARKER_TYPE_MAYBE_BLOCKING_SYSCALL markers and system call parameters
+     * and return values for kernel locks (SYS_futex on Linux) using the function tracing
+     * markers #TRACE_MARKER_TYPE_FUNC_ID, #TRACE_MARKER_TYPE_FUNC_ARG, and
+     * #TRACE_MARKER_TYPE_FUNC_RETVAL with an identifier equal to
+     * #TRACE_FUNC_ID_SYSCALL_BASE plus the system call number (or its bottom 16 bits
+     * for 32-bit marker values).  These identifiers are not stored in the function
+     * list file (drmemtrace_get_funclist_path()).
+     *
+     * The #TRACE_MARKER_TYPE_FUNC_RETVAL for system calls is either 0 (failure) or
+     * 1 (success).
+     */
+    OFFLINE_FILE_TYPE_KERNEL_SCHED = 0x800,
```

Comment:
We may identify more data required for scheduling in future. So better to use a name that describes the actual data added to the trace, not what that data enables; something like `OFFLINE_FILE_TYPE_BLOCKING_SYSCALL_AND_FUTEX` or just `OFFLINE_FILE_TYPE_BLOCKING_SYSCALL`

Reply:
Hmm, we could do KERNEL_SCHED_2.  I liked KERNEL_SCHED tying these together.  But maybe lumping futex into BLOCKING_SYSCALLS is ok.  All right, will go with BLOCKING_SYSCALLS.